modules/transactionpool/transactionpool_test.go

```diff
@@ -336,3 +336,327 @@ func TestFeeEstimation(t *testing.T) {
+		err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graph[0]})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	_, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add all of the transactions in each graph into the tpool, one transaction
+	// at a time, interweaved, chaotically.
+	for i := 1; i < len(graphs[0]); i++ {
+		for j := 0; j < len(graphs); j++ {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graphs[j][i]})
+			if err != nil {
+				t.Fatal(err, i, j)
+			}
+		}
+	}
+
+	// Mine blocks until the tpool is gone.
+	for tpt.tpool.transactionListSize > 0 {
+		_, err := tpt.miner.AddBlock()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestHeapFees(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	tpt, err := createTpoolTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tpt.Close()
+
+	// Mine a few more blocks to get some extra funding.
+	for i := 0; i < 5; i++ {
+		_, err := tpt.miner.AddBlock()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	// Make a bunch of transactions with high fees. A bunch with low fees.
+	// See that the next block has all high fee txns.
+	// Iterate over block txns and add up fees
+	// Prepare a bunch of outputs for a series of graphs to fill up the
+	// transaction pool.
+	coinFrac := types.SiacoinPrecision
+
+	numGraphs := 110
+	graphFund := coinFrac.Mul64(12210)
+	var outputs []types.SiacoinOutput
+	for i := 0; i < numGraphs; i++ {
+		outputs = append(outputs, types.SiacoinOutput{
+			UnlockHash: types.UnlockConditions{}.UnlockHash(),
+			Value:      graphFund,
+		})
+	}
+	txns, err := tpt.wallet.SendSiacoinsMulti(outputs)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Mine the graph setup in the consensus set so that the graph outputs are
+	// transaction sets.
+	_, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	finalTxn := txns[len(txns)-1]
+
+	// For each output, create 250 transactions
+	var graphs [][]types.Transaction
+	for i := 0; i < numGraphs; i++ {
+		var edges []types.TransactionGraphEdge
+		var cumFee types.Currency
+		for j := 0; j < numGraphs; j++ {
+			fee := coinFrac.Mul64(uint64((j + 1)))
+			cumFee = cumFee.Add(fee)
+			edges = append(edges, types.TransactionGraphEdge{
+				Dest:   j + 1,
+				Fee:    fee,
+				Source: 0,
+				Value:  fee,
+			})
+		}
+		for k := 0; k < numGraphs; k++ {
+			fee := coinFrac.Mul64(uint64(k + 1)).Div64(2)
+			cumFee = cumFee.Add(fee)
+			edges = append(edges, types.TransactionGraphEdge{
+				Dest:   k + 251,
+				Fee:    fee,
+				Source: k + 1,
+				Value:  fee,
+			})
+		}
+
+		graph, err := types.TransactionGraph(finalTxn.SiacoinOutputID(uint64(i)), edges)
+		if err != nil {
+			t.Fatal(err)
+		}
+		graphs = append(graphs, graph)
+
+	}
+
+	for _, graph := range graphs {
+		err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graph[0]})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	block, err := tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
should merge this with the block above and explain that you are confirming the head of each graph so that the rest of the transactions in the graph will be viewed as individual transactions.